elements/rh-chip/rh-chip.ts

```diff
@@ -0,0 +1,111 @@
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { property } from 'lit/decorators/property.js';
+import { query } from 'lit/decorators/query.js';
+import { colorContextConsumer, type ColorTheme } from '../../lib/context/color/consumer.js';
+import { consume } from '@lit/context';
+
+import { context, type RhChipGroupContext } from './context.js';
+
+import '@rhds/elements/rh-icon/rh-icon.js';
+
+import styles from './rh-chip.css';
+
+export class ChipCheckedEvent extends CustomEvent<{ checked: boolean }> {
+  constructor(checked: boolean) {
+    super('chip-checked', {
+      bubbles: true,
+      composed: true,
+      detail: { checked },
+    });
+  }
+}
+
+/**
+ * Chip
+ * A chip is a clickable surface that acts like a checkbox.
+ * @summary Creates a checkable surface
+ * @fires {ChipCheckedEvent} chip-checked - when chip is checked/unchecked
+ * @csspart chip - The main chip container
+ * @slot - Place the label of the checkbox here
+ */
+@customElement('rh-chip')
+export class RhChip extends LitElement {
+  static readonly styles = [styles];
+
+  /**
+   * Sets color theme based on parent context
+   */
+  @colorContextConsumer() private on?: ColorTheme;
+
+  /**
+   * Whether the chip is checked.
+   */
+  @property({ type: Boolean, reflect: true }) checked = false;
+
+  /**
+   * Whether the chip is disabled.
+   */
+  @property({ type: Boolean, reflect: true }) disabled = false;
+
+  /**
+   * Set a custom string for the input's `name` attribute. Defaults to `chip-checkbox`.
+   */
+  @property({ reflect: true, attribute: 'name' }) chipName?: string;
+
+  /**
+   * Set a custom string for the input's `value` attribute. Defaults to `on`.
+   */
+  @property({ reflect: true, attribute: 'value' }) chipValue?: string;
+
+  @query('input') private chipInput!: HTMLInputElement;
+
+  @consume({ context, subscribe: true })
+  @property({ attribute: false })
+  private ctx?: RhChipGroupContext;
+
+  render() {
+    const { on = 'light' } = this;
+    const attrName = this.chipName ? this.chipName : 'chip-checkbox';
+    const { size } = this.ctx ?? {};
+    return html`
+      <label part="chip" class=${classMap({ on: true, [on]: true, [`size-${size}`]: !!size })}>
+        <slot></slot>
+        <input type="checkbox"
+               name=${attrName}
+               value=${ifDefined(this.chipValue)}
+               @change=${this.#onChecked}
+               @keydown=${this.#handleKeydown}
+               ?checked=${!this.disabled && this.checked}
+               aria-disabled=${!!this.disabled}>
+        <rh-icon id="close-icon" set="ui" icon="close-circle"></rh-icon>
+      </label>
+    `;
+  }
+
+  #onChecked(event: Event) {
+    if (this.disabled) {
+      event.preventDefault();
+      (event.target as HTMLInputElement).checked = this.checked;
+      return;
+    }
+
+    this.checked = this.chipInput.checked;
+    this.dispatchEvent(new ChipCheckedEvent(this.checked));
```

Comment:
```suggestion
    event.stopPropagation();
    if (this.disabled) {
      event.preventDefault();
      return;
    }

    if (this.dispatchEvent(new ChipChangeEvent(this.checked))) {
      this.checked = this.chipInput.checked;
    }
```

handle the state of the input in the template.

Reply:
Handled this in [f23cbc3](https://github.com/RedHat-UX/red-hat-design-system/pull/2198/commits/f23cbc3570d556244212cb887ef86bfc53d9ed84). Kept: 

```ts
(event.target as HTMLInputElement).checked = this.checked;
```

...because, without it, users can check a disabled chip with their keyboard / space bar. In its current form, disabled chips are never checked per design specs.

Reply:
Re-revewing the design specs, looks like a checked and disabled chip has been added in the last 18 hours. I will work on adding this state.

Reply:
> Re-revewing the design specs, looks like a checked and disabled chip has been added in the last 18 hours. I will work on adding this state.

@adamjohnson Yeah, I thought you needed those. Also, please check your Slack messages about needing a chip that clears multiple `Selected` chips.